nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-2-6-processors/src/main/java/org/apache/nifi/processors/kafka/pubsub/PublisherLease.java

```diff
@@ -221,6 +222,19 @@ void publish(final FlowFile flowFile, final RecordSet recordSet, final RecordSet
         }
     }
 
+    private String resolveMessageKeyIdNeeded(MessageKeyResolver keyResolver, FlowFile flowFile, Record record, PropertyValue messageKeyField) {
+        if(keyResolver != null) {
+            final String key = keyResolver.apply(flowFile, record, messageKeyField);
+            if(key != null) {
+                logger.debug("Message key resolved: key={}, messageKeyField={}", key, messageKeyField);
+                return key;
+            } else {
+                logger.warn("Message key resolver configured, but no key resolved! messageKeyField={}, record={}", messageKeyField, record);
```

Comment:
I think I would avoid making this a WARN message and just make it DEBUG. I can see a use case where a null value is intended as a key. In this case, logs would be spammed with non-stop WARN logs and the processor may always show bulletins.